rotkehlchen/tests/api/test_users.py

```diff
@@ -631,6 +631,46 @@ def test_user_login(rotkehlchen_api_server, username, db_password, data_dir):
     assert users_data[username] == 'loggedin'
     assert users_data['another_user'] == 'loggedout'
 
+    # Pretend that a user db upgrade is ongoing, so that we can test that on user login
+    # while an upgrade is ongoing, they will be prompted to resume from backup
+    with rotki.data.db.user_write() as write_cursor:
+        rotki.data.db.set_setting(  # Pretend that an upgrade was started
+            write_cursor=write_cursor,
+            name='ongoing_upgrade_from_version',
+            value=33,
+        )
+
+    # Logout again
+    data = {'action': 'logout'}
+    response = requests.patch(
+        api_url_for(rotkehlchen_api_server, 'usersbynameresource', name=username),
+        json=data,
+    )
+    assert_simple_ok_response(response)
+    assert rotki.user_is_logged_in is False
+    users_data = check_user_status(rotkehlchen_api_server)
+    assert len(users_data) == 2
+    assert users_data[username] == 'loggedout'
+    assert users_data['another_user'] == 'loggedout'
+
+    # Now let's try to login (without a consent to resume from backup)
+    data = {'password': db_password, 'sync_approval': 'unknown', 'async_query': True, 'resume_from_backup': False}  # noqa: E501
+    response = requests.post(
+        api_url_for(rotkehlchen_api_server, 'usersbynameresource', name=username),
+        json=data,
+    )
+    task_id = assert_ok_async_response(response)
+    response_data = wait_for_async_task(rotkehlchen_api_server, task_id)
+    assert_error_async_response(
+        response_data=response_data,
+        contained_in_msg='Resume from a backup or quit login',
+        status_code=HTTPStatus.MULTIPLE_CHOICES,
+        result_exists=True,
+    )
+    # Notice: Any login after this point will require a user db backup to be resumed from.
+    # If you need to login without a backup write your test case before the point at which
+    # the user db upgrade status is set to ongoing.
```

Comment:
Why add this comment and not just resume from a backup normally? This also should be part ofthe test otherwise the test is incomplete.